scanapp/views.py

```diff
@@ -161,3 +123,79 @@ def post(self, request):
                 }
             )
         )
+
+
+class UrlScanView(FormView):
+    template_name = 'scanapp/urlscan.html'
+    form_class = UrlScanForm
+
+    def post(self, request, *args, **kwargs):
+        form = self.form_class(request.POST)
+
+        if form.is_valid():
+            # get the URL from the form
+            url = request.POST['url']
+
+            logger = logging.getLogger(__name__)
+
+            if parse_url(url) == 1:
+
+                gitparser = GitURL(url)
+
+                url = gitparser.to_git()
+
+                home_path = expanduser("~")
+                os.chdir(home_path)
+
+                if (str(request.user) == 'AnonymousUser'):
+                    path = 'media/AnonymousUser/url/'
+                    user = None
+                else:
+                    path = 'media/user/' + str(request.user) + '/url/'
+                    user = request.user
+
+                scan_start_time = datetime.now()
+                subprocess.call(['mkdir', '-p', path])
```

Comment:
Are you serious? Calling a subprocess to create a directory? did you check the Python doc?